enhancements/ptp-ha.md

```diff
@@ -0,0 +1,267 @@
+ABMC (Alternate BMC is required).
+
+## Scenarios
+
+### Multiple NICs
+
+This case assumes that over the that the Virtual Distributed Unit (vDU) hosts
+several NICs supporting Boundary Clock Function. The synchronization source is
+selected by *ptp4l*
+
+On Openshift 4.10 the PTP operator supports running multiple NICs with the
+following dual nic with two ptp4l configurations, but no HA on failure.
+
+```text
+        ┌─────────────────────────┐
+        │                         │
+        │                         │
+        │    Grand Master         │
+        │                         │
+        │                         │
+        └───────┬───────────┬─────┘
+                │           │
+                │           │
+                │           │
+         ┌──────┼───────────┼────┐
+         │ vDU  │           │    │
+         │ ┌────┴───┐  ┌────┴──┐ │
+         │ │        │  │       │ │
+         │ └─────┬──┘  └────┬──┘ │
+         │  NIC1 │      NIC2│    │
+         └───────┼──────────┼────┘
+                 │          │
+                 │          │
+          ┌──────┼──────────┼───┐
+          │      │          │   │
+          │      │       ┌──┘   │
+          │      │       │      │
+         ┌┼┬────┬┼┐     ┌┼┬────┬┼┐
+         ├─┘    └─┤     ├─┘    └─┤
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │ RU#0   │     │ RU#1   │
+         └────────┘     └────────┘
+```
+
+HA on failure would require several NICs connection at the Remote Unit (RU)
+level as well, so they would be able to load balance between the two boundary
+clocks within the boundary clock.
+
+### Multiple interfaces on one NIC
+
+- Should this be covered?
+
+## Motivation
+
+### Goals
+
+- Support multiple PTP sources within the PTP operator
+- Add an additional field in the CR regarding stratum priority
+
+### Non-Goals
+
+- This EP doesn't consider CPU pinning assigned to the different processes.
+
+## Proposal
+
+### Reconcile Loop
+
+The PTP Operator's reconcile loop would also take care of checking whether the
+PTP clock has failed, should the HA field be on the CRD.
+
+#### What does failure mean?
+
+Failure would mean that the sync with the upstream clock is lost, and that the
+local ptp daemon is on freerun. It would have its sync lost bit by bit, even
+though it may still be working for some time.
+
+Once such situation is identified, We would need to kill the phc2sys daemon.
+The secondary nic will configured with its `phc2sys` crd config set to zero
+so it doesn't start with any config.
+
+Would this be a new controller on the operator? Some new parameter on the
+reconcile loop? Opening this EP so we can iterate on this.
+
+### Workflow Description
+
+### API Extensions
+
+### User Stories
+
```

Comment:
can some initial story titles be added?  it would help with the understanding of the work